ivi/rohdeschwarz/rohdeschwarzBaseRFSigGen.py

```diff
@@ -0,0 +1,174 @@
+
+    def __init__(self, *args, **kwargs):
+        self.__dict__.setdefault('_instrument_id', '')
+        
+        super(rohdeschwarzBaseRFSigGen, self).__init__(*args, **kwargs)
+
+        self._rf_rms_voltage_level = 0.0
+
+        # frequency limit in Hertz
+        self._frequency_low = 9e3
+        self._frequency_high = 1100e6        
+        # rf level limit in dBm
+        self._rf_level_low = -120.0
+        self._rf_level_high = 19
+        # rf rms voltage level limit in Volt
+        self._rf_rms_voltage_level_low = 223.61e-9
+        self._rf_rms_voltage_level_high = 1.993
+
+        self._identity_description = "Rohde&Schwarz generic IVI RF signal generator driver"
+        self._identity_identifier = ""
+        self._identity_revision = ""
+        self._identity_vendor = ""
+        self._identity_instrument_manufacturer = "Rohde&Schwarz"
+        self._identity_instrument_model = ""
+        self._identity_instrument_firmware_revision = ""
+        self._identity_specification_major_version = 1
+        self._identity_specification_minor_version = 0
+        self._identity_supported_instrument_models = list(['SMC100A'])
+
+        self._add_property('rf.rms_voltage_level',
+                        self._get_rf_rms_voltage_level,
+                        self._set_rf_rms_voltage_level,
+                        None,
+                        ivi.Doc("""
+                        Rms voltage level of the instrument.
+                        """))
+
+    def _initialize(self, resource = None, id_query = False, reset = False, **keywargs):
+        "Opens an I/O session to the instrument."
+        
+        super(rohdeschwarzBaseRFSigGen, self)._initialize(resource, id_query, reset, **keywargs)
+        
+        # interface clear
+        if not self._driver_operation_simulate:
+            self._clear()
+        
+        # check ID
+        if id_query:
+            id = self.identity.instrument_model
+            id_check = self._instrument_id
+            id_short = id[:len(id_check)]
+            if id_short != id_check:
+                raise Exception("Instrument ID mismatch, expecting %s, got %s", id_check, id_short)
+        
+        # reset
+        if reset:
+            self.utility_reset()
+        
+    def _get_rf_frequency(self):
+        "Reads the frequency of the generated RF output signal. The unit is Hertz"
+        self._rf_frequency = float(self._ask("FREQ?"))
+        return self._rf_frequency
+    
+    def _set_rf_frequency(self, value):
+        "Specifies the frequency of the generated RF output signal. The unit is Hertz"
+        value = float(value)
+        if value < self._frequency_low or value > self._frequency_high:
+            raise ivi.OutOfRangeException()
+        self._write("FREQ %e HZ" % value)
+        self._rf_frequency = value
+    
+    def _get_rf_level(self):
+        """
+        return set value in dBm of rf output level containing offset.
+        """
```

Comment:
Add
if not self._driver_operation_simulate and not self._get_cache_valid():

Reply:
i dont plan to use caching because read is fast..., not sure why simulate is needed